saleor/graphql/product/utils.py

```diff
@@ -111,3 +117,28 @@ def clean_variant_sku(sku: Optional[str]) -> Optional[str]:
     if sku:
         return sku.strip() or None
     return None
+
+
+def update_ordered_media(ordered_media):
+    for order, media in enumerate(ordered_media):
+        media.sort_order = order
+        try:
+            media.save(update_fields=["sort_order"])
+        except DatabaseError as e:
+            msg = (
+                "Cannot update media for instance: %s. "
+                "Possible updating not existing object. "
+                "Details: %s.",
+                media,
+                str(e),
+            )
+            logger.warning(msg)
+
+            raise ValidationError(
+                {
+                    "order": ValidationError(
```

Comment:
Are you sure, that the `order` field is OK here? I would say, that we should include here a `media`. Also it happens in the loop: 
- Shouldn't we return a list of exception here?
- Don't we need a transaction here? To make sure that the whole operation finished with success or fail? Currently, there is a risk of partial update.

Reply:
Good point!

Reply:
[fixed](https://github.com/saleor/saleor/pull/9225/commits/237af39e573412557186f5d8febf2a63e845d270)